src/toil_scripts/gatk_germline/germline.py

```diff
@@ -1,688 +1,570 @@
-
-
-def return_input_paths(job, work_dir, ids, *filenames):
-    """
-    Given one or more filenames, reads the file from the fileStore and
-    copies it (or makes a hard link) into the working directory.
-
-    :param job: Job instance
-    :param work_dir: working directory, str
-    :param ids: dictionary of shared file ids
-    :param filenames: remaining arguments are filenames
-
-    :returns: list of paths to files
+def download_and_run(job, uuid, url, config, rg_line=None):
     """
-    paths = OrderedDict()
-    for filename in filenames:
-        if not os.path.exists(os.path.join(work_dir, filename)):
-            file_path = job.fileStore.readGlobalFile(ids[filename], os.path.join(work_dir, filename))
-        else:
-            file_path = os.path.join(work_dir, filename)
-        paths[filename] = file_path
-
-    return paths.values()
-
+    Downloads reference files and runs the GATK best practices germline pipeline for a single
+    samples.
 
-def write_to_filestore(job, work_dir, ids, *filenames):
+    :param JobFunctionWrappingJob job: Toil Job instance
+    :param str uuid: Unique identifier for the sample
+    :param str url: URL to BAM file
+    :param Namespace config: Configuration options for pipeline
+    :param str rg_line: RG line for BWA alignment. Default is None.
+    :return: None
     """
-    Given one or more file names in working directory, writes
-    files to filestore and stores the filestore promise in a dictionary
+    get_shared_files = job.wrapJobFn(download_shared_files, config).encapsulate()
+    samples = [(uuid, url, rg_line)]
+    run_pipeline = Job.wrapJobFn(run_gatk_germline_pipeline, samples, get_shared_files.rv())
+    get_shared_files.addChild(run_pipeline)
 
-    :param job: Job instance
-    :param work_dir: working directory
-    :param ids: shared file promises, dict
-    :param filenames: remaining arguments are keys for ids
 
-    :returns: shared ids dictionary
+def run_gatk_germline_pipeline(job, samples, config):
     """
-    for filename in filenames:
-        ids[filename] = job.fileStore.writeGlobalFile(os.path.join(work_dir, filename))
-    return ids
-
-
-def read_from_filestore_hc(job, work_dir, ids, *filenames):
-    """
-    Reads file from fileStore and writes it to working directory.
-
-    :param job: Job instance
-    :param work_dir: working directory
-    :param ids: shared file promises, dict
-    :param filenames: remaining arguments are filenames
-    """
-    for filename in filenames:
-        if not os.path.exists(os.path.join(work_dir, filename)):
-            job.fileStore.readGlobalFile(ids[filename], os.path.join(work_dir, filename))
-
+    Configures and runs the GATK germline pipeline with filtering.
 
-def move_to_output_dir(work_dir, output_dir, *filenames):
-    """`
-    Moves files from the working directory to the output directory.
+    VQSR is done if the run_vqsr parameter is set to True or there are at least
```

Comment:
RE: not autodetecting. It's pretty common to call variants on targeted sequencing data where the number of sites covered is so small that even with large sample sizes (e.g., 500+ samples), you don't cover enough sites for VQSR to not flip out.